FinancialTracker/ViewControllers/HomeViewController.swift

```diff
@@ -51,95 +54,112 @@ class HomeViewController: UIViewController {
         }
     }
     
+    private func expenseOrIncome(start: Date, end: Date) -> [Transaction] {
+        var data: [Transaction] = []
+        if expenseOrIncomeSegmentedControl.selectedSegmentIndex == 0 {
+            data = start.transactionBetweenTwoDates(till: end, data: authManager?.currentUser?.expenses ?? [])
+        } else {
+            data = start.transactionBetweenTwoDates(till: end, data: authManager?.currentUser?.incomes ?? [])
+        }
+        return data
+    }
+    
+    @IBAction func expenseOrIncomeControlDidChange(_ sender: UISegmentedControl) {
+        dividerControlDidChange(expenseDividerSegmentedControl)
+    }
+    
     @IBAction func dividerControlDidChange(_ sender: UISegmentedControl) {
         switch sender.selectedSegmentIndex {
         case 0:
```

Comment:
You can put the `case 0` (the `Int`s) in an descriptive enum which has rawValue of type `Int` so that your cases are more descriptive.

Reply:
In other words, you will wrap the `Int` cases in an enum.